recirq/engine_utils.py

```diff
@@ -372,3 +375,39 @@ async def worker():
     await queue.join()
     for wjob in worker_jobs:
         wjob.cancel()
+
+
+def _get_current_time():
+    return datetime.datetime.now()
+
+
+def get_available_processors(processor_names: List[str]):
+    """Checks the reservation status of the processors and returns a list of
+    processors that are available to run on at the present time.
+    """
+    project_id = os.environ['GOOGLE_CLOUD_PROJECT']
+    engine = cirq.google.get_engine()
+    available_processors = []
+    for processor_name in processor_names:
+        processor_id = get_processor_id_by_device_name(processor_name)
+        if processor_id is None:
+            # Skip the check if this is a simulator.
+            continue
+        processor = engine.get_processor(processor_id)
+        for time_slot in processor.get_schedule():
+            try:
+                time_slot = EngineTimeSlot.from_proto(time_slot)
+            # Parsing the end_time of the last time range in the schedule might give throw an error.
+            except ValueError:
+                continue
+            # Ignore time slots that do not contain the current time.
+            if time_slot.start_time < _get_current_time() < time_slot.end_time:
```

Comment:
lets set _get_current_time() at the beginning of the loop for efficiency and to make sure the calculation is consistent.

Reply:
Done